test/support/bad_connection_plug.ex

```diff
@@ -0,0 +1,32 @@
+defmodule NervesHubLinkCommon.Support.BadConnectionPlug do
+  @moduledoc """
+  Simulates a connection where no HTTP data is sent after headers
+  """
+
+  @behaviour Plug
+
+  import Plug.Conn
+
+  @impl Plug
+  def init(options), do: options
+
+  @impl Plug
+  def call(conn, _opts) do
+    conn
+    |> put_resp_header("content-length", "10000000000001")
+    |> send_chunked(200)
+    |> slow_stream()
+  end
+
+  # this stream will obviously never complete
+  def slow_stream(conn) do
+    conn =
+      for byte <- 0..10_000_000_000_001 do
+        Process.sleep(5000)
+        {:ok, conn} = chunk(conn, <<byte::8>>)
+        conn
+      end
```

Comment:
I'm guessing this totally doesn't matter, but it seems like you want a reduce here rather than a list comprehension.

```suggestion
    Enum.reduce(0..10_000_000_000_001, conn, fn byte, conn ->
        Process.sleep(5000)
        {:ok, conn} = chunk(conn, <<byte::8>>)
        conn
      end)
```

Reply:
ha. you are totally right.